src/bidiMapper/domains/context/BrowsingContextImpl.ts

```diff
@@ -1012,6 +1014,99 @@ export class BrowsingContextImpl {
 
     await this.#cdpTarget.toggleNetworkIfNeeded(enableNetwork);
   }
+
+  async locateNodes(
+    params: BrowsingContext.LocateNodesParameters
+  ): Promise<BrowsingContext.LocateNodesResult> {
+    // TODO: test sandboxing.
+    const realm = await this.getOrCreateSandbox(params.sandbox);
+    if (params.maxNodeCount !== undefined) {
+      // TODO: implement.
+      throw new UnsupportedOperationException(`maxNodeCount is not supported`);
+    }
+    if (params.ownership === ResultOwnership.Root) {
+      // TODO: implement.
+      throw new UnsupportedOperationException(
+        `ownership:root is not supported`
+      );
+    }
+    if (params.serializationOptions !== undefined) {
+      // TODO: implement.
+      throw new UnsupportedOperationException(
+        `serializationOptions is not supported`
+      );
+    }
+    if (params.startNodes !== undefined) {
+      // TODO: implement.
+      throw new UnsupportedOperationException(
+        `serializationOptions is not supported`
+      );
+    }
+
+    switch (params.locator.type) {
+      case 'css':
+        return await this.#locateNodesByCssSelector(
+          realm,
+          params.locator.value
+        );
+      default:
+        throw new UnsupportedOperationException(
+          `locateNodes does not support ${params.locator.type} locator type.`
+        );
+    }
+  }
+
+  async #locateNodesByCssSelector(
```

Comment:
How about we move this to the `WindowRealm` class?

Reply:
I guess the implementation should be implemented via `DOM.querySelectorAll`, as otherwise the elements form the shadow tree will not be searched. And if so, it is more relevant to the BrowsingContext rather then window realm.